src/System.Private.CoreLib/shared/System/Text/UnicodeEncoding.cs

```diff
@@ -31,6 +31,8 @@ public class UnicodeEncoding : Encoding
         // Unicode version 2.0 character size in bytes
         public const int CharSize = 2;
 
+        // endianness-based bit pattern mask.
+        static readonly ulong highLowPatternMask = ((ulong) 0xd800d800d800d800 | (BitConverter.IsLittleEndian ? (ulong) 0x0400000004000000 : (ulong) 0x0000040000000400)); 
```

Comment:
I have missed this one - this one is actually regressing performance. I will take care of fixing it together with https://github.com/dotnet/corefx/issues/24192#issuecomment-394087942